imports/plugins/core/checkout/client/templates/checkout/payment/methods/cards.js

```diff
@@ -42,3 +42,67 @@ Template.corePaymentMethods.helpers({
     return self;
   }
 });
+Template.walletPayment.onCreated(function () {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.setDefault({
+      details: {balance: 0}
+    });
+    this.subscribe("transactionDetails", Meteor.userId());
+    const transactionInfo = Wallets.find().fetch();
+    if (transactionInfo.length > 0) {
+      this.state.set("details", transactionInfo[0]);
+    }
+  });
+});
+
+Template.walletPayment.helpers({
+  balance: () => {
+    return Template.instance().state.get("details").balance;
+  }
+});
+
+Template.walletPayment.events({
+  "click #pay-with-wallet": (event) => {
+    event.preventDefault();
+    const balance = Template.instance().state.get("details").balance;
+    const cartAmount = Number(Cart.findOne().cartTotal());
+    const currency = Shops.findOne().currency;
+    if (cartAmount > balance) {
+      Alerts.toast("Insufficient balance", "error");
+      return false;
+    }
+    transactionId = Random.id();
+    Meteor.call("wallet/transaction", Meteor.userId(), {
+      amount: cartAmount,
+      date: new Date(),
+      orderId: transactionId,
+      transactionType: "Debit"
+    }, (err, res) => {
+      if (res) {
+        const paymentMethod = {
+          processor: "Wallet",
+          storedCard: "",
+          method: "credit",
+          transactionId,
+          currency: currency,
+          amount: cartAmount,
+          status: "passed",
+          mode: "authorize",
+          createdAt: new Date(),
+          transactions: []
+        };
+        const theTransaction = {
+          amount: cartAmount,
+          transactionId,
+          currency: currency
+        };
+        paymentMethod.transactions.push(theTransaction);
+        Meteor.call("cart/submitPayment", paymentMethod);
+        Alerts.toast("Payment Successful", "success");
+      } else {
+        Alerts.toast("Oops!!, an error occured, please try again", "error");
```

Comment:
Does this alert hide or remains static?

consider using autoHide in your toast

Reply:
It hides

Reply:
okay